components/script/dom/gpuadapter.rs

```diff
@@ -237,35 +237,37 @@ impl GPUAdapterMethods for GPUAdapter {
 }
 
 impl AsyncWGPUListener for GPUAdapter {
-    fn handle_response(&self, response: WebGPUResponseResult, promise: &Rc<Promise>) {
+    fn handle_response(&self, response: Option<WebGPUResponseResult>, promise: &Rc<Promise>) {
         match response {
-            Ok(WebGPUResponse::RequestDevice {
-                device_id,
-                queue_id,
-                descriptor,
-            }) => {
-                let device = GPUDevice::new(
-                    &self.global(),
-                    self.channel.clone(),
-                    &self,
-                    Heap::default(),
-                    descriptor.features,
-                    descriptor.limits,
+            Some(response) => match response {
+                Ok(WebGPUResponse::RequestDevice {
                     device_id,
                     queue_id,
-                    descriptor.label.unwrap_or_default(),
-                );
-                self.global().add_gpu_device(&device);
-                promise.resolve_native(&device);
-            },
-            Err(e) => {
-                warn!("Could not get GPUDevice({:?})", e);
-                promise.reject_error(Error::Operation);
-            },
-            _ => {
-                warn!("GPUAdapter received wrong WebGPUResponse");
-                promise.reject_error(Error::Operation);
+                    descriptor,
+                }) => {
+                    let device = GPUDevice::new(
+                        &self.global(),
+                        self.channel.clone(),
+                        &self,
+                        Heap::default(),
+                        descriptor.features,
+                        descriptor.limits,
+                        device_id,
+                        queue_id,
+                        descriptor.label.unwrap_or_default(),
+                    );
+                    self.global().add_gpu_device(&device);
+                    promise.resolve_native(&device);
+                },
+                Err(e) => {
+                    unreachable!("Failed to get a response from the constellation ({})", e)
+                },
+                _ => {
+                    warn!("GPUAdapter received wrong WebGPUResponse");
+                    promise.reject_error(Error::Operation);
+                },
             },
+            None => promise.resolve_native(&None::<GPUDevice>),
```

Comment:
Here the `None` should be `unreachable`, because it's only the `RequestAdapter` message that can return `None`, all other request imply an initial adapter was acquired, therefore WebGPU is available and `None` should never be subsequently returned.